src/node/cli.ts

```diff
@@ -325,13 +325,37 @@ export const parse = (
     args._.push(arg)
   }
 
+  // If a cert was provided a key must also be provided.
+  if (args.cert && args.cert.value && !args["cert-key"]) {
+    throw new Error("--cert-key is missing")
+  }
+
   logger.debug("parsed command line", field("args", args))
 
   return args
 }
 
-export async function setDefaults(args: Args): Promise<Args> {
-  args = { ...args }
+export interface DefaultedArgs extends ConfigArgs {
```

Comment:
Everything in `Args` is optional so this new interface reflects that some of these properties will definitely exist after setting the defaults. 

The practical benefit is that you can directly use things like `someFunc(args.host)` instead of having to do something like `args && someFunc(args.host) || throw new Error("host wasn't set")` or `someFunc(args.host!)`.